pkg/util/k8sutil/events.go

```diff
@@ -283,3 +283,12 @@ func NewOperatorEngineOpsAlertEvent(reason string, apiObject APIObject) *Event {
 	event.Message = fmt.Sprintf("Event OperatorEngineOpsAlert raised, investigation needed: %s", reason)
 	return event
 }
+
+// NewCannotSetArchitectureARM64Event creates an even of type CannotSetArchitectureARM64.
+func NewCannotSetArchitectureARM64Event(apiObject runtime.Object, memberId string) *Event {
+	event := newDeploymentEvent(apiObject)
+	event.Type = core.EventTypeWarning
+	event.Reason = "Can not set architecture ARM64"
+	event.Message = fmt.Sprintf("Can not apply ARM64 arch for member %s. It is not supported for ArangoDB < 3.10.0", memberId)
```

Comment:
The function named like a general "can't set architecture event" but actually it reports the reason: ArangoD version incompatibility. Maybe pass the message or message part through func parameter?

Reply:
+1